Formula/purescript-language-server.rb

```diff
@@ -0,0 +1,21 @@
+require "language/node"
+
+class PurescriptLanguageServer < Formula
+  desc "Node-based Language Server Protocol server for PureScript"
+  homepage "https://github.com/nwolverson/purescript-language-server"
+  url "https://registry.npmjs.org/purescript-language-server/-/purescript-language-server-0.16.6.tgz"
+  sha256 "b7a705c51c7bad8cc1c2e8d9e827a57140e90c2e290f1381e64fb16418453e99"
+  license "MIT"
+
+  depends_on "node"
+  depends_on "purescript"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "Connection input stream is not set.", shell_output("#{bin}/purescript-language-server 2>&1", 1)
```

Comment:
We _can_ actually pretend to be an LSP client and communicate with the LSP server here, but I think this is sufficient.

Reply:
You can also check with the other LSP tests so that some LSP JSON testing can be tested out.

Reply:
I went ahead and did exactly that. Initially, I tried the `shell_output` approach (used in `bash-language-server` for instance), but the `purescript-language-server` invocation failed with exit code 1 for whatever reason. I then tried the `Open3.popen3` approach (used in `ansible-language-server`), and it worked just fine.